src/cluster.c

```diff
@@ -6076,7 +6076,7 @@ int clusterRedirectBlockedClientIfNeeded(client *c) {
         (c->btype == BLOCKED_LIST ||
          c->btype == BLOCKED_ZSET ||
          c->btype == BLOCKED_STREAM ||
-         c->btype == BLOCKED_MODULE))
+         (c->btype == BLOCKED_MODULE && moduleClientIsBlockedOnKeys(c))))
```

Comment:
> The client which is blocked for other reasons may also need to be redirected when the cluster is down. So we may also need to deal with these clients.
> 
> https://github.com/redis/redis/blob/157454ffa4388143f9d68753d0a94bc684010367/src/cluster.c#L6058-L6065
> 
> Do you think so?


@oranagra　Ｗhat's your opinion about this？

Reply:
@qetu3790 do you wanna handle that in this PR (and add a test)? or in a followup PR?

Reply:
I guess that the following code wants to unblock the client when the cluster is down.
https://github.com/redis/redis/blob/24b67d5520ca062d2c0ed432112fd3c26ceb3daa/src/cluster.c#L6083-L6090
Module commands have their particularity. These commands written by users can block clients for other reasons(not for keys).Like `tests/modules/blockonbackground.so`. So if we want to unblock these clients, we may change codes like this  and add some test code in `tests/unit/cluser.tcl`. Otherwise these clients will continue to be blocked even if the cluster fails.
```suggestion
         c->btype == BLOCKED_MODULE))
``` 
Do we want to notify these clients which is blocked for other reasons? If we treat the cluster as a entirety, we may need to unblock them. And i can try to add the test codes.But yossigo has approved the PR. I don't know how to do the next. I need your suggestion

Reply:
@qetu3790 how about my last commit, does that look good?
wanna try add a test for it?